viewing_party/party.py

```diff
@@ -1,23 +1,187 @@
+    unique_user_movies = [movie for movie in watched_user if movie not in dupes]
+    return unique_user_movies
+
+
+
+def get_friends_unique_watched(user_data):
+    watched_user = user_data["watched"]
+    friends = user_data["friends"]
+    unique_friend_movies = []
+    for friend in friends:
+        watched_friend = friend["watched"]
+        for movie in watched_friend:
+            if movie not in watched_user:
+                if movie not in unique_friend_movies:
+                    unique_friend_movies.append(movie)     
+    return unique_friend_movies 
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+#create get_available_recs function
+# within movie dict there is a 'host' key with streaming services as values
+# create empty list for recommended movies
+# for each movie in friend's watched 
+# if movie not in user's watched
+# and movie's 'host' value in subcriptions
+# append movie to recommended movies list
+def get_available_recs(user_data):
+    recommended_movies = []
+    watched_user = user_data["watched"]
+    subscriptions = user_data["subscriptions"]
+    friends = user_data["friends"]
+    for friend in friends:
+        watched_friend = friend["watched"]
+        for movie in watched_friend:
+            if movie not in watched_user:
+                if movie["host"] in subscriptions:
+                    recommended_movies.append(movie)
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+# create a get new rec by genre function
+# create empty list called recommended_movies
+# create a variable called "favorite genre"
+# assign result from most watched genre function to variable
+# if movie in friends' watched list
+# and if movie not in user watched list
+# and movie["genre"] == favorite genre
+# append movie to recommended_movies list
+def get_new_rec_by_genre(user_data):
+    recommended_movies = []
+    favorite_genre = get_most_watched_genre(user_data)
+    friends = user_data["friends"]
+    watched_user = user_data["watched"]
+    for friend in friends:
+        watched_friend = friend["watched"]
+        for movie in watched_friend:
+            if movie not in watched_user and movie["genre"] == favorite_genre:
+                recommended_movies.append(movie)
+    return recommended_movies
+
+# create get_rec_from_favorites function with user_data parameter
+# user_data has a "favorites" dict with list of user's favorite movies as strings
+# create an empty list called recommended_movies
+# create "unique_movies" variable to hold output from get_unique watched function
+# create "favorites" variable to access favorites list
+# loop through movies in watched_user
+# if movie in unique movies and favorites
+# append movie to recommended_list
+def get_rec_from_favorites(user_data):

```

Comment:
Another great solution. I would challenge you to think about if using `get_unique_watched()` results in the fastest solution. Is there a better way to do this?